test/StateL.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+var FL = require('fantasy-land');
+
+
+var StateL = function StateL(f) {
+  if (!(this instanceof StateL)) {
+    return new StateL(f);
+  }
+  this._f = f;
+};
+
+StateL.prototype['@@type'] = 'sanctuary-type-classes/StateL';
+
+StateL[FL.of] = function(x) {
+  return StateL(function(s) {
+    return {
+      accum: s,
+      value: x
+    };
+  });
+};
+
+StateL.prototype[FL.map] = function(f) {
+  var fn = this._f;
```

Comment:
It's a bit awkward to have two functions named `f` and `fn`. It reminds me a bit of `String#substr` and `String#substring`, something which has bothered me for some time. Could we name this `_f` instead? At least then we'd be mirroring the field name.